packages/exo/src/exo-tools.js

```diff
@@ -6,59 +6,150 @@ import {
   isAwaitArgGuard,
+  isRawGuard,
   getAwaitArgGuardPayload,
   getMethodGuardPayload,
   getInterfaceGuardPayload,
   getCopyMapEntries,
 } from '@endo/patterns';
 
+import { GET_INTERFACE_GUARD } from './get-interface.js';
+
 /** @typedef {import('@endo/patterns').Method} Method */
 /** @typedef {import('@endo/patterns').MethodGuard} MethodGuard */
+/** @typedef {import('@endo/patterns').MethodGuardPayload} MethodGuardPayload */
 /**
  * @template {Record<PropertyKey, MethodGuard>} [T=Record<PropertyKey, MethodGuard>]
  * @typedef {import('@endo/patterns').InterfaceGuard<T>} InterfaceGuard
  */
-/** @typedef {import('@endo/patterns').InterfaceGuardKit} InterfaceGuardKit */
 
 const { quote: q, Fail } = assert;
 const { apply, ownKeys } = Reflect;
 const { defineProperties, fromEntries } = Object;
 
+/**
+ * A method guard, for inclusion in an interface guard, that does not
+ * enforce any constraints of incoming arguments or return results.
+ */
+const RawMethodGuard = M.call().rest(M.raw()).returns(M.raw());
+
+const REDACTED_RAW_ARG = '<redacted raw arg>';
+
 /**
  * A method guard, for inclusion in an interface guard, that enforces only that
  * all arguments are passable and that the result is passable. (In far classes,
- * "any" means any *passable*.) This is the least possible enforcement for a
- * method guard, and is implied by all other method guards.
+ * "any" means any *passable*.) This is the least possible non-raw
+ * enforcement for a method guard, and is implied by all other
+ * non-raw method guards.
+ */
+const PassableMethodGuard = M.call().rest(M.any()).returns(M.any());
+
+/**
+ * @typedef {object} RedactConfig
+ * @property {number} declaredLen
+ * @property {boolean} hasRestArgGuard
+ * @property {Pattern} paramsPattern
+ * @property {boolean} rawRestArgs
+ * @property {number[]} redactedIndices
  */
-const MinMethodGuard = M.call().rest(M.any()).returns(M.any());
 
 /**
  * @param {Passable[]} syncArgs
- * @param {MethodGuardPayload} methodGuardPayload
+ * @param {RedactConfig} redactConfig
  * @param {string} [label]
  * @returns {Passable[]} Returns the args that should be passed to the
  * raw method
  */
-const defendSyncArgs = (syncArgs, methodGuardPayload, label = undefined) => {
-  const { argGuards, optionalArgGuards, restArgGuard } = methodGuardPayload;
-  const paramsPattern = M.splitArray(
-    argGuards,
-    optionalArgGuards,
-    restArgGuard,
-  );
-  mustMatch(harden(syncArgs), paramsPattern, label);
-  if (restArgGuard !== undefined) {
+const defendSyncArgs = (syncArgs, redactConfig, label = undefined) => {
+  const {
+    declaredLen,
+    hasRestArgGuard,
+    paramsPattern,
+    rawRestArgs,
+    redactedIndices,
+  } = redactConfig;
+
+  // Only manipulate the original args if our params pattern didn't contain any
+  // raw values.
+  let redactedArgs = syncArgs;
+  if (rawRestArgs) {
+    // Don't harden the rest args.
+    redactedArgs = syncArgs.slice(0, Math.min(syncArgs.length, declaredLen));
```

Comment:
I don't think this `Math.min` is doing anything; [`slice`](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.prototype.slice) already clamps its second argument to the length of the array.
```suggestion
    redactedArgs = syncArgs.slice(0, declaredLen);
```